### PR TITLE
Replace redundant <form> tags with <div>s

### LINKED ISSUE #1143: Pressing return in search box reloads page
### Current Behavior

While filtering a table (e.g. users, devices, teams or projects) if you type in the search box and then press enter it reloads the page (and looses the search term)

### Expected Behavior

Not reload the page

### Steps To Reproduce

Open any page with a table + search option
Start typing in the search box
Press enter

### Environment

- FlowForge version: 1.0.0
- Node.js version: 16.x
- npm version: 8.x
- Platform/OS: NA
- Browser: Chrome

### PR DESCRIPTION
**Pull request by joepavitt**
Closes #1143

### CODE REVIEW

File: frontend/src/pages/team/Settings/General.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-    <form class="space-y-6">
+    <div class="space-y-6">
```
Comment (Pezmc) [inline frontend/src/pages/team/Settings/General.vue:2]: For this (and the some of other settings changes) semantically `form` does feel like a better fix than div. This part of the page does indeed contain a form, even if we're not using HTML form handling.

Comment (joepavitt) [reply frontend/src/pages/team/Settings/General.vue:2]: very fair point - in most cases where we are using form - it isn't a form. Happy to revisit each of them and sanity check their usage.

Comment (knolleary) [reply frontend/src/pages/team/Settings/General.vue:2]: Whilst I do agree (which is why I used them originally), this is also a pragmatic fix for the issue as we don't rely on any of the browser-defined `<form>` behaviours.

We should aspire to restore `<form>` where appropriate - but will need to solve the UX issue around unexpected/inappropriate form submission.

**Review by knolleary — APPROVED** [review]

**Comment by joepavitt** [comment]
@knolleary just a note as I've just pushed post-approval, that I've just reverted some based on @Pezmc's feedback

File: frontend/src/pages/team/Settings/General.vue (hunk 1 above)
Comment (Pezmc) [reply frontend/src/pages/team/Settings/General.vue]: I'm :+1: to merge this as is and follow up (or create an issue) to restore the forms where appropriate.

**Review by Pezmc — APPROVED** [review]

**Pull request merged**